systemdspawner/systemdspawner.py

```diff
@@ -276,6 +288,9 @@ async def start(self):
             else:
                 working_dir = self._expand_user_vars(self.user_workingdir)
 
+        if self.cpu_weight:
+            env["CPU_WEIGHT"] = str(self.cpu_weight)
+
```

Comment:
To add a new env var here sets a new precendece for systemdspawner of passing environment variables related to the config set.

I suspect this can add to maintenance burden long term, so I suggest this is removed unless there is a solid motivation for retaining it.

I know some information about system constraints can be detected in a standardized ways, for example as done [in dask/distributed](https://github.com/dask/distributed/blob/7926ea620c24f5c47429ebc88380a048edc6982c/distributed/system.py#L25-L34). So if this is relevnat information for the user environment, maybe it can be captured in a similar way?